File: selection_sort/selection_sort.py
""" Selection sort
"""
from random import randint

def selection_sort(array:list):
    """ Sort an array using Selection Sort """
    size = len(array)
    for i in range(size):
        min_val, min_idx = array[i], None
        for k in range(i+1, size):
            if array[k] < min_val:
                min_val, min_idx = array[k], k
        if min_val < array[i]:
            array[i], array[min_idx] = array[min_idx], array[i]

if __name__ == "__main__":
    nterms = 10
    a = [randint(0, 99) for _ in range(nterms)]
    print(a)
    selection_sort(a)
    print(a)
    
    

File: selection_sort/selection_sort.go
// Selection Sort
package main

import (
    "fmt"
    "math/rand"
    "time"
)

// Order the array using the Selection Sort method
func selection_sort(array []int) {
    var min_val, min_idx, j int;
    for i := range array {
        min_val = array[i];
        for j=i+1; j<len(array); j++ {
            if array[j] < min_val {
                min_val, min_idx = array[j], j
            }
        }
        if array[i] > min_val {
            array[i], array[min_idx] = array[min_idx], array[i]
        }
    }
}

// The main function
func main() {
    rand.Seed(time.Now().UnixNano())
    // Creates a random array
    arr := make([]int, 15);
    for i:=0; i<len(arr); i++ {
        arr[i] = rand.Intn(99)
    }
    fmt.Println(arr)
    selection_sort(arr)
    fmt.Println(arr)
}
